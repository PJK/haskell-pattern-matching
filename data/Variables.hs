module Variables where

data MyBool
    = True
    | False

f :: MyBool -> MyBool -> Integer
f True y = 4
f x y = 3