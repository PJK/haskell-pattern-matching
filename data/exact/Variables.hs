module Variables where

-- f :: MyBool -> MyBool -> Integer
-- f True y = 4
-- f x y = 3
--
-- incompleteF :: MyBool -> MyBool -> Integer
-- incompleteF True y = 4
-- incompleteF x False = 3

data MyTrinary = A | B | C

-- incompleteFT :: MyTrinary -> MyTrinary -> Integer
-- incompleteFT A y = 4
-- incompleteFT x B = 3

data MyMaybe = Sum MyTrinary | Nope

extract :: MyMaybe -> Bool -> MyTrinary
extract (Sum x) False = x
extract (Sum C) True = C
