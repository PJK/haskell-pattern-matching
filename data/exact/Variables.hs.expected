{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "extract",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "A",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "True",
                                []
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "B",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "True",
                                []
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        },
                        {
                            "tag": "WildcardPattern",
                            "contents": []
                        }
                    ]
                ]
            }
        ]
    ]
}
